examples/features/gracefulstop/README.md

```diff
@@ -0,0 +1,31 @@
+# Graceful Stop
+
+This example demonstrates how to gracefully stop a gRPC server using
+Server.GracefulStop().
+
+## How to run
+
+Start the server which will listen to incoming gRPC requests as well as OS
+interrupt signals (SIGINT/SIGTERM). After receiving interrupt signal, it calls
+`s.GracefulStop()` to gracefully shut down the server. If graceful shutdown
+doesn't happen in time, server is stopped forcefully.
+
+```sh
+$ go run server/main.go
+```
+
+In a separate terminal, start the client which will send multiple requests to
+the server with some delay between each request.
+
+```sh
+$ go run client/main.go
+```
+
+Use Ctrl+C or SIGTERM to signal the server to shut down.
+
+The server begins a graceful stop:
+- It finishes ongoing requests.
+- Rejects new incoming requests.
```

Comment:
I'm not sure how precise we want to be here, but this isn't _exactly_ what happens, FYI.

The server sends a GOAWAY on the connection, which the client sees, and then the client will stop sending requests on the connection.  If the client ignores the GOAWAY, then the server will reject new streams, but that shouldn't actually happen in any properly functioning HTTP/2 system.

What you wrote might be good enough for a basics explanation though.

Reply:
Thanks. Now i am just restricting the description to completing the in-flight requests because we are not demonstrating sending new request while server is shutting down.